posts/2024-01-31-Performance-Doris_Amoakohene/index.qmd

```diff
@@ -0,0 +1,274 @@
+
+In data.table, the term “Performance Regression” usually means a drop in both time metrics and memory metrics, and that is how we will be  using the term here.
+
+
+# **Why do we run Performance Tests on commits?**
+
+Running performance tests on commits helps maintain a high-performance standard for the package, detect and fix performance regressions, optimize code, validate performance improvements, and ensure consistent performance over time. It is an essential practice to deliver a performant and reliable package to end-users.
+
+# **What are the Performance Tests?**
+The goal of our atime Performance Tests is to gather memory and responsiveness time metrics while simulating the full range of member interactions with the data.table repository
+
+## In atime code, there are five main parts:
+
+1. `pkg.path`: This variable represents the path to the package being benchmarked. It specifies the location of the `data.table` package on your system.
+
+2. `N`: This variable determines the number of iterations for the benchmarking process. It is a sequence of numbers that define different data sizes to test the performance of the operation.
+
+3. `setup`: This section contains the setup code for generating the dataset used in the benchmarking process. 
+
+4. `expr`: This section contains the expression that represents the operation being benchmarked. It uses the `data.table::`[.data.table`` syntax to perform the  operation on the dataset. 
+
+5. `...` : This section specifies the different versions of the data.table packages  that will be tested. It includes three versions: "Before," "Regression," and "Fixed." Each version is associated with a specific commit id.
+
+The result from running the atime versions will be a list of the seconds.limit (numeric input param) and timings (data table of results).
+
+Lastly, I run a github action. The action defines test.list as a list with names corresponding to different tests. Each element of the test.list should be a list with named arguments N, setup, expr, which was  passed as arguments in your atime::atime_versions test. For further elaboration on the process of performing asymptotic time testing using the atime package, please refer to [this ](https://github.com/marketplace/actions/r-asymptotic-testing)
+
+## We run the full performance, Pull Request (PR):
+1. Before the issue is made (Before)
+2. when the PR is first submitted (Regression)
+3. when the PR is merged to the destination branch(Fixed)
+
+# **APPROACH**
+1. To begin, conduct the atime test for the different code branches (before regression, regression, fix regression) to identify potential performance issues. Here is an example of how to perform the [atime test](https://github.com/DorisAmoakohene/Efficiency-and-Preformance-Test.RData.table)
+
+NB: Set up the necessary environment and dependencies, ensuring that the data.table package and the atime package are installed and loaded.
```

Comment:
please use full English words instead of abbreviations like NB

Reply:
okay